README.md

```diff
@@ -99,6 +99,19 @@ entrypoint:
 $ docker run -v $(pwd):/app -v '/app/xain_fl.egg-info' xain-fl-dev coordinator
 ```
 
+You can pass a custom config file and custom initial weights by
+mounting volumes:
+
+```shell
+docker run \
+  -v $(pwd)/custom_config.toml:/custom_config.toml \
+  -v $(pwd)/custom_initial_weights.npy:/custom_initial_weights.npy \
+  -v $(pwd):/app \
+  -v '/app/xain_fl.egg-info' \
+  xain-fl-dev \
+  coordinator --config /custom_config.toml
+```
+
```

Comment:
Just one small additional thing:

Since we don't have the `cli.py` file anymore, we need to remove or update the `Running the Coordinator locally`  section in the `README.md`

Reply:
When I try to run this example I get the following error:
```
Traceback (most recent call last):
  File "/usr/local/bin/coordinator", line 11, in <module>
    load_entry_point('xain-fl', 'console_scripts', 'coordinator')()
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 490, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2853, in load_entry_point
    return ep.load()
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2444, in load
    return self.resolve()
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2450, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/app/xain_fl/__main__.py", line 41, in <module>
    main()
  File "/app/xain_fl/__main__.py", line 23, in main
    config = Config.load(args.config)
  File "/app/xain_fl/config/schema.py", line 345, in load
    with open(path, "r") as f:
IsADirectoryError: [Errno 21] Is a directory: '/custom_config.toml'
```
Can we maybe just use the default config?

```
coordinator --config docker/dev/xain-fl.toml 
```

Reply:
This section explains how to pass a custom config to the container by mounting a config file and a `.npy` file in the container:

```shell
  -v $(pwd)/custom_config.toml:/custom_config.toml \
  -v $(pwd)/custom_initial_weights.npy:/custom_initial_weights.npy \
```

In this command, if `$(pwd)/custom_config.toml` and `$(pwd)/custom_initial_weights.npy` don't exist on your host, docker creates empty directories in the container. This is why you get this error message.

Try this out:

```
cp docker/dev/initial_weights.npy custom_initial_weights.npy
cp docker/dev/xain-fl.toml custom_config.toml
sed -i  's#^initial_weights.*#initial_weights = /custom_initial_weights.npy#g' custom_config.toml
docker run \
  -v $(pwd)/custom_config.toml:/custom_config.toml \
  -v $(pwd)/custom_initial_weights.npy:/custom_initial_weights.npy \
  -v $(pwd):/app \
  -v '/app/xain_fl.egg-info' \
  xain-fl-dev \
  coordinator --config /custom_config.toml
```

> Can we maybe just use the default config?

The previous command in the README does that already, but I want to document how to pass a custom config as well, because I'm pretty sure someone will need it, and it's not that obvious.

Reply:
I pushed a few changes. First I made the README clearer about what this command does.

Also we now log proper error and exit if we cannot open the config file or read it.

For instance if the file is actually a directory (your case):
```
❯ sudo rm -r custom_config.toml custom_initial_weights.npy
❯ docker run \
  -v $(pwd)/custom_config.toml:/custom_config.toml \
  -v $(pwd)/custom_initial_weights.npy:/custom_initial_weights.npy \
  -v $(pwd):/app \
  -v '/app/xain_fl.egg-info' \
  xain-fl-dev \
  coordinator --config /custom_config.toml

{
  "event": "Invalid config: /custom_config.toml is a directory",
  "level": "error"
}
```

If the file contains invalid TOML:
```
❯ cp docker/dev/initial_weights.npy custom_initial_weights.npy                                                                          
cp docker/dev/xain-fl.toml custom_config.toml                                                                                           
sed -i  's#^initial_weights.*#initial_weights = /custom_initial_weights.npy#g' custom_config.toml                                       
docker run \
  -v $(pwd)/custom_config.toml:/custom_config.toml \
  -v $(pwd)/custom_initial_weights.npy:/custom_initial_weights.npy \ 
  -v $(pwd):/app \
  -v '/app/xain_fl.egg-info' \
  xain-fl-dev \
  coordinator --config /custom_config.toml


{
  "event": "Invalid config: failed to decode /custom_config.toml: This float doesn't have a leading digit (line 9 column 1 char 224)",
  "level": "error"
}
```